resources/prosody-plugins/mod_muc_room_metadata.lua

```diff
@@ -0,0 +1,217 @@
+-- This module implements a generic metadata storage system for rooms.
+--
+-- This module is added under the main virtual host domain
+-- It needs a room metadata muc component
+--
+-- VirtualHost "jitmeet.example.com"
+--     modules_enabled = {
+--         "muc_room_metadata"
+--     }
+--     room_metadata_muc = "metadata.jitmeet.example.com"
+--     main_muc = "muc.jitmeet.example.com"
+--
+-- Component "metadata.jitmeet.example.com" "muc"
+--     restrict_room_creation = true
+--     storage = "memory"
+--     admins = { "focusUser@auth.jitmeet.example.com" }
+--     muc_room_locking = false
+--     muc_room_default_public_jids = true
+
+local jid_node = require 'util.jid'.node;
+local json = require 'util.json';
+local st = require 'util.stanza';
+
+local util = module:require 'util';
+local is_healthcheck_room = util.is_healthcheck_room;
+
+local COMPONENT_IDENTITY_TYPE = 'room_metadata';
+local FORM_KEY = 'muc#roominfo_jitsimetadata';
+
+local main_muc_component_config = module:get_option_string('main_muc');
+if main_muc_component_config == nil then
+    module:log('error', 'breakout rooms not enabled missing main_muc config');
+    return ;
+end
+local room_metadata_muc_component_config = module:get_option_string('room_metadata_muc', 'metadata.'..module.host);
+
+module:depends('jitsi_session');
+
+local component_muc_service;
+local main_muc_service;
+
+
+-- Utility functions
+
```

Comment:
Yeah maybe leave a comment that is an optimization step so clients don't need to be notified via separate message about it on join.

Reply:
Done.